src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -370,34 +356,26 @@ def activeTask(doc, spec):
         # for each expectedRSEs not in localCopyOfCurrentRSEs
         for rse in set(expectedRSEs).difference(localCopyOfCurrentRSEs):
             # check whether the RSE has enough available space available for that
-            if isinstance(rucioClient, Client):
-                generator = rucioClient.get_rse_usage(rse)
-            else:
-                generator = rucioClient.getRSEUsage(rse)
+            # taken from logic of RSEQuotas, see https://bit.ly/3kDJZmO
+            # but here we use explicitly rse parameter
+            records = rucioClient.getAccountUsage(rucioAccount, rse)
             enoughSpace = False
-            for rec in generator:
-                # use nodeUsage provided in spec to get rse space info
-                if nodeUsage and rse in nodeUsage:
-                    rseUsage = nodeUsage[rse]
-                    if rseUsage['bytes_remaining'] > marginSpace:
-                        enoughSpace = True
-                        break
-                else:
-                    # we'll use total - used space to exceed pileupSize with some margin
-                    if rec['total'] - rec['used'] - pileupSize > marginSpace:
-                        enoughSpace = True
-                        break
+            for rec in records:
+                # each record has form
+                # {"rse": ..., "bytes_limit": ..., "bytes": ..., "bytes_remaining": ...}
+                if rec['rse'] != rse:
```

Comment:
Given that the call to `getAccountUsage` above is already specific to a single RSE, I would say this if statement is not really needed.

Reply:
As I look into the wrapper API, I think we should actually change the default value type returned to an empty list:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/Rucio/Rucio.py#L155

if you agree, feel free to make that change as well.

Reply:
yes, I changed Rucio code as well